zerver/views/invite.py

```diff
@@ -46,16 +46,20 @@ def invite_users_backend(
     invite_expires_in_minutes: Optional[int] = REQ(
         json_validator=check_none_or(check_int), default=INVITATION_LINK_VALIDITY_MINUTES
     ),
-    invite_as: int = REQ(json_validator=check_int, default=PreregistrationUser.INVITE_AS["MEMBER"]),
+    invite_as: int = REQ(
+        json_validator=check_int_in(
+            list(PreregistrationUser.INVITE_AS.values()),
+            message=_("Must be invited as a valid type of user"),
```

Comment:
The engineering is nice on this `message` feature! But from a practical perspective, I think it might be better to not add it, and instead just revert to the default "Invalid invite_as" text. My reasoning is basically that this particular field is unlikely to be one that you send an invalid value for via the API, and so there's no reason to have our translators deal with an extra string that doesn't have material utility beyond the default one.

So my recommendation is we skip the `message` feature and just revert to the default "Invalid ..." string here.